### PR TITLE
gcd using euclidean algorithm || mathematics

### LINKED ISSUE #301: documentation for Finding GCD using Euclidean algorithm || math
## 🚀 Feature

(A clear and concise description of what the feature is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

GCD using Euclidean algorithm

## Pitch

GCD using Euclidean algorithm

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #301 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #301

### CODE REVIEW

**Comment by aaadddiii** [comment]
@iamrajiv  @HarshCasper  please review my pr

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+## Complexity Analysis
+
+- Time complexity - O(`log(min(a,b))`)
+- Space complexity - O(`1`)
+
+
+## Credits
+
+-   [vybhav](https://github.com/vybhav72954) for [implementation in c++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/GCD_In_C.cpp)
```
Comment (vybhav72954) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:120]: I am not the author of the script.  Use git blame to determine the actual author.

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+slug: /mathematics/gcd-euclidean-algorithm
+---
+
+## Introduction
+
+The GCD of two number is the largest integer that divides each of the integers such that their remainder is zero . The Euclidean algorithm is an efficient way for finding the GCD of `2` numbers. It is named after the greek mathematician Euclid
+
+## Explanation
+
+Consider two numbers `a = 333` and `b = 84` <br/>
```
Comment (vybhav72954) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:12]: Pls remove the break tags.

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
+---
+title: GCD using Euclidean algorithm
+slug: /mathematics/gcd-euclidean-algorithm
+---
+
+## Introduction
+
+The GCD of two number is the largest integer that divides each of the integers such that their remainder is zero . The Euclidean algorithm is an efficient way for finding the GCD of `2` numbers. It is named after the greek mathematician Euclid
```
Comment (vybhav72954) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:8]: Grammar check required

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+int gcd(int a,int b)
+{ //base case
+  if (a==0)
+     return b;
+  //recursive case
+  else
+     return gcd(b%a,a);
+}
+
+//initializing a and b.
```
Comment (vybhav72954) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:55]: A whitespace after comment block is recommended. 

Like this

```
// Notice the whitespace

```

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    return 0;
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```python
+def gcd_of_nums(x,y):
+    #choose the smaller number
```
Comment (vybhav72954) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:74]: Here as well

File: docs/mathematics/gcd-euclidean-algorithm.mdx (hunk 1 above)
Comment (aaadddiii) [reply docs/mathematics/gcd-euclidean-algorithm.mdx:120]: It's showing you as the author when I did git blame

**Comment by aaadddiii** [comment]
@vybhav72954  I have made the changes, can you please review it

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@aaadddiii Resolve the merge conflict. Also, I don't think `leaf-nodes-bst.mdx` belongs to this PR.

**Comment by aaadddiii** [comment]
@iamrajiv  I have done the changes can you review it

**Review by iamrajiv — APPROVED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  Can you please review and merge my pr

**Comment by aaadddiii** [comment]
@iamrajiv  can you please review my pr

**Review by iamrajiv — APPROVED** [review]

File: docs/mathematics/gcd-euclidean-algorithm.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+     return gcd(b%a,a);
+}
+
+// initializing a and b.
+// Here a is always the smaller number and
+// b is always the bigger number between the two.
+
+int main()
+{
+    cout<<"Enter a and b for finding GCD"<<endl;
```
Comment (siddharth25pandey) [inline docs/mathematics/gcd-euclidean-algorithm.mdx:65]: Write proper statement
**Enter a and b for finding GCD** is vague statement
Write something like this, Enter two numbers........

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Ayushjain2205 Do the suggested changes

**Review by ankitaggarwal23 — COMMENTED** [review]
Thank you for your valuable time.😊

**Pull request closed without merging**